src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1216,6 +1228,12 @@ public Component getTableCellRendererComponent(JTable table, Object value,
 
             return this;
         }
+
+        public double formatToDoubleValue(long len) {
+            DecimalFormat df = new DecimalFormat("0.0");
```

Comment:
DecimalFormat("0.0") does not provide locale-specific group separator, which may come to play.
for example "1,000 TB"

Reply:
also, it always formats numbers to have one decimal point, even for whole numbers (10.0)
is this intended functionality?  should "0.#" be used instead?

Reply:
Group separator may come into picture if file size is in terms of TB, otherwise for files having sizes in bytes , KB, MB and so on will be converted to next higher unit if size > 999.

e.g. 
1000 bytes - 1KB
1000 KB - 1 MB and so on...

So, I guess group separator may not come into play.

Reply:
> also, it always formats numbers to have one decimal point, even for whole numbers (10.0) is this intended functionality? should "0.#" be used instead?

JFileChooser is not showing one decimal point for whole numbers. 
e.g. 1000 KB file is shown as 1 MB. 

![image](https://user-images.githubusercontent.com/107542245/181587024-b05786c6-2c92-4230-9fb2-bcb5fdda1082.png)